File: go/internal/scoring/testdata/structural-v1/java/StateCoherentBefore.java
final class StateCoherentBefore {
  private int total;

  int advance(int delta) {
    if (delta < 0) return total;
    total += delta;
    if (total > 100) total = 100;
    return total;
  }

  int advanceAgain(int delta) {
    if (delta < 0) return total;
    total += delta;
    if (total > 100) total = 100;
    return total;
  }
}

File: go/internal/scoring/testdata/structural-v1/go/state_coherent_before.go
package calibration

type CoherentBefore struct{ total int }

func (c *CoherentBefore) Advance(delta int) int {
	if delta < 0 {
		return c.total
	}
	c.total += delta
	if c.total > 100 {
		c.total = 100
	}
	return c.total
}

func (c *CoherentBefore) AdvanceAgain(delta int) int {
	if delta < 0 {
		return c.total
	}
	c.total += delta
	if c.total > 100 {
		c.total = 100
	}
	return c.total
}
